From: Andrew Morton <akpm@linux-foundation.org>
Subject: drivers/gpu/drm/dp/drm_dp.c: "fix" build

drivers/gpu/drm/dp/drm_dp.c: In function 'drm_dp_get_adjust_request_post_cursor':
drivers/gpu/drm/dp/drm_dp.c:59:27: error: array subscript 10 is outside array bounds of 'const u8[6]' {aka 'const unsigned char[6]'} [-Werror=array-bounds]
   59 |         return link_status[r - DP_LANE0_1_STATUS];
      |                ~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
drivers/gpu/drm/dp/drm_dp.c:210:51: note: while referencing 'link_status'
  210 | u8 drm_dp_get_adjust_request_post_cursor(const u8 link_status[DP_LINK_STATUS_SIZE],
      |                                          ~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [scripts/Makefile.build:288: drivers/gpu/drm/dp/drm_dp.o] Error 1
make[3]: *** [scripts/Makefile.build:614: drivers/gpu/drm/dp] Error 2
make[2]: *** [scripts/Makefile.build:614: drivers/gpu/drm] Error 2
make[1]: *** [scripts/Makefile.build:614: drivers/gpu] Error 2
make: *** [Makefile:1971: drivers] Error 2

Cc: Keith Packard <keithp@keithp.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 drivers/gpu/drm/dp/drm_dp.c |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/drivers/gpu/drm/dp/drm_dp.c~drivers-gpu-drm-dp-drm_dpc-fix-build
+++ a/drivers/gpu/drm/dp/drm_dp.c
@@ -56,7 +56,8 @@ struct dp_aux_backlight {
 /* Helpers for DP link training */
 static u8 dp_link_status(const u8 link_status[DP_LINK_STATUS_SIZE], int r)
 {
-	return link_status[r - DP_LANE0_1_STATUS];
+//	return link_status[r - DP_LANE0_1_STATUS];
+	return 0;
 }
 
 static u8 dp_get_lane_status(const u8 link_status[DP_LINK_STATUS_SIZE],
_
